app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,22 @@
+class PagesController < ApplicationController
+
+# before_action :authenticate_user!
+
+  def search
+    if params[:search].blank?
+      render 'search'
+    end
+
+    if Page.count > 0
```

Comment:
It feels like the `Page` model should be responsible for handling this searching.  You can create a class method on `Page`:
```ruby
class Page
  def self.term_search(terms)
    return nil if count <= 0

    result = search do
      fulltext(terms) do
        highlight :body
      end
    end

    return result.hits
  end
end
```

Then your controller method might look like:

```ruby
def search
  return render 'search' if params[:search].blank?

  @results = Page.term_search(params[:search])
  
  if @results.nil?
    @message = "Start your first search"
  end
end
```